internal/server/server.go

```diff
@@ -35,26 +38,47 @@ func requestHandler(ctx *fasthttp.RequestCtx) {
 	if host == "" {
 		ctx.Response.SetStatusCode(404)
 		ctx.Response.SetBodyString(fmt.Sprintf("Domain %s not found", ctx.Host()))
-	} else {
-		req.SetHost(host)
+	} else { // the route exists so we should have a deployment attached to the service
+		origin := store.GetRouteOrigin(host)
+		req.SetHost(origin)
 
+		// First try
 		if err := httpClient.Do(req, res); err != nil {
```

Comment:
May not be an error on first time if you get a 503. I think err only occurs when something seriously wrong happens at the socket level. Will have to test / double check this though. But my guess is that error will always == nil and u will get a 503

Reply:
The 5XX should be when the origin return an error.
In this case, the pod is down so the client cannot call the url, therefore error is not nil.
This is confirmed by the tests I am running but will need more.